xc/device.mli

```diff
@@ -21,6 +21,20 @@ exception Device_not_found
 
 exception Cdrom
 
+module Profile: sig
+	type t = Qemu_trad | Qemu_upstream_compat | Qemu_upstream
+	val fallback : t
+	module Name: sig
+		val qemu_trad: string
+		val qemu_upstream_compat: string
+		val qemu_upstream: string
+		val all: string list
+	end
+	val wrapper_of: t -> string
+	val string_of : t -> string
+	val of_string : string -> t
```

Comment:
This can fail. How does it fail?

Reply:
It falls back to Qemu_trad, which is also of type t (see Profile.fallback). This fallback value is established in the design paper and depends on the stage.